tests/90jira/SYN-205.pl

```diff
@@ -11,14 +11,25 @@
       ->then( sub {
          my ( $room_id ) = @_;
 
-         await_event_for( $invitee, filter => sub {
-            my ( $event ) = @_;
-            return unless $event->{type} eq "m.room.member" and
-                          $event->{room_id} eq $room_id and
-                          $event->{state_key} eq $invitee->user_id and
-                          $event->{content}{membership} eq "invite";
+         await_sync($invitee, check => sub {
+            my ( $sync_body ) = @_;
+            log_if_fail $sync_body;
+            my $room =  $sync_body->{rooms}{invite}{$room_id};
+            return unless $room;
```

Comment:
it'd be nice to make the falsey return value explicit, rather than relying on undef to be falsey.

```suggestion
            return 0 unless $room;
```